internal/cache/util/util.go

```diff
@@ -99,3 +101,25 @@ func IsCacheHandleInvalid(readErr error) bool {
 		strings.Contains(readErr.Error(), ErrInSeekingFileHandleMsg) ||
 		strings.Contains(readErr.Error(), ErrInReadingFileHandleMsg)
 }
+
+func CheckCacheDirectoryPermissions(dirPath string) (bool, error) {
```

Comment:
(feel free to ignore if you don't agree)
Nit: Since this method is returning bool, I think it is better to name it IsCacheDirectoryWriteable or something similar

Reply:
I also agree a method returning bool should start with is, but modifying this method's behaviour as per below comment and naming it accordingly
https://github.com/GoogleCloudPlatform/gcsfuse/pull/1639#discussion_r1465080889